### PR TITLE
Add docs about plugins section in Pulumi.yaml

### PR DESCRIPTION
**Pull request by Frassle**
See https://github.com/pulumi/pulumi/pull/10146 for implementation

### CODE REVIEW

**Comment by harribo11** [comment]
Should we mark providers/languages/analyzers as optional?

**Review by harribo11 — APPROVED** [review]

File: themes/default/content/docs/reference/pulumi-yaml.md [hunk 1]
```diff
@@ -48,3 +48,9 @@ It contains the following required and optional attributes:
         - `description`: (optional) a description for the config setting.
         - `default`: (optional) the default value of the config setting - which will be presented to the user as a default.
         - `secret`: (optional) if `true` indicates that this configration value should be marked as secret.
+
+- `plugins`: (optional) allows the overriding of plugin selection to make use of plugins not installed into the global plugin cache
```
Comment (susanev) [inline themes/default/content/docs/reference/pulumi-yaml.md:52]:
```suggestion
- `plugins`: (optional) allows the overriding of plugin selection to make use of plugins not installed into the global plugin cache.
```

File: themes/default/content/docs/reference/pulumi-yaml.md [hunk 2]
```diff
@@ -48,3 +48,9 @@ It contains the following required and optional attributes:
...
+  - `providers`/`analyzers`/`languages`: (optional) Each entry has a list of plugins underneath it.
```
Comment (susanev) [inline themes/default/content/docs/reference/pulumi-yaml.md:53]:
```suggestion
  - `providers`/`analyzers`/`languages`: (optional) each entry has a list of plugins underneath it.
```

would only saying `list of plugins.` make sense?

**Review by susanev — APPROVED** [review]

File: themes/default/content/docs/reference/pulumi-yaml.md (hunk 2 above)
Comment (Frassle) [reply themes/default/content/docs/reference/pulumi-yaml.md:53]: Yes, but given the slight odd format of this I was erring on a longer description so people trigger that this is multiple entries like:
```
plugins:
  providers:
  - name: aws
    path: ../aws
  analyzers:
  - name: company_policy
    path: ./policy
```

**Pull request merged**